recursion.tex

```diff
@@ -228,23 +247,24 @@ \subsection{Implementing recursive queries}\label{sec:datalog}
 %$\code{O} = \fix{x}{R(\code{I}, x)}$.
 %\end{proof}
 
-Note that this loop is not guaranteed to terminate.  But if a program
-does converge, the above construction will find the least fixed point.
+Note that this loop is not guaranteed to terminate.  But if the
```

Comment:
We should be more precise here. It _is_ guaranteed to terminate for the class of Datalog programs that have a fixed point, e.g., programs with stratified negation. So it's not just pure luck.